first-party/jib-layer-filter-extension-maven/src/test/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtensionTest.java

```diff
@@ -291,4 +326,115 @@ public void testExtendContainerBuildPlan_complex() throws JibPluginExtensionExce
+          ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noParent()
+      throws JibPluginExtensionException {
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().build();
+    Configuration cfg = new Configuration();
+    cfg.setCreateParentLayers(true);
+
+    when(mavenProject.getParent()).thenReturn(null);
+
+    try {
+      JibLayerFilterExtension extension = new JibLayerFilterExtension();
+      extension.setDependencyResolver(projectDependenciesResolver);
+
+      extension.extendContainerBuildPlan(buildPlan, null, Optional.of(cfg), mavenData, logger);
+      fail();
+    } catch (JibPluginExtensionException ex) {
+      assertEquals(JibLayerFilterExtension.class, ex.getExtensionClass());
+      assertEquals("Try to get parent dependencies, but project has no parent.", ex.getMessage());
+    }
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_noParentDependencies()
+      throws JibPluginExtensionException {
+    FileEntriesLayer layer1 = buildLayer("", Arrays.asList("/app/libs/foo-1.0.0.jar"));
+    ContainerBuildPlan buildPlan = ContainerBuildPlan.builder().addLayer(layer1).build();
+    Configuration cfg = new Configuration();
+    cfg.setCreateParentLayers(true);
+
+    when(dependencyResolutionResult.getDependencies()).thenReturn(Collections.emptyList());
+
+    JibLayerFilterExtension extension = new JibLayerFilterExtension();
+    extension.setDependencyResolver(projectDependenciesResolver);
+
+    ContainerBuildPlan newPlan =
+        extension.extendContainerBuildPlan(buildPlan, null, Optional.of(cfg), mavenData, logger);
+
+    assertEquals(1, newPlan.getLayers().size());
+    FileEntriesLayer newLayer1 = (FileEntriesLayer) newPlan.getLayers().get(0);
+    assertEquals("", newLayer1.getName());
+    assertEquals(layer1.getEntries(), newLayer1.getEntries());
+  }
+
+  @Test
+  public void testExtendContainerBuildPlan_createParentLayers_withParentDependencies()
+      throws JibPluginExtensionException {
+    FileEntriesLayer layer1 =
+        buildLayer(
+            "layer1",
+            asList(
+                "/app/libs/foo-1.0.0.jar",
+                "/app/libs/bar-2.0.0.jar",
+                "/app/libs/wrong-version-2.0.0.jar"));
+    FileEntriesLayer layer2 = buildLayer("layer2", asList("/app/libs/blah-3.0.0.jar"));
+
+    ContainerBuildPlan buildPlan =
+        ContainerBuildPlan.builder().addLayer(layer1).addLayer(layer2).build();
+    Configuration cfg = new Configuration();
+    cfg.setCreateParentLayers(true);
+
+    Dependency fooDependency = mockDependency("foo", "1.0.0");
+    Dependency blahDependency = mockDependency("blah", "3.0.0");
+    // If the version does not match, the dependency must not be moved to parent layer
+    Dependency wrongVersionDependency = mockDependency("wrong-version", "1.0.0");
```

Comment:
For readability, how about
```suggestion
    Dependency directDependency = mockDependency("direct-lib", "1.0.0");
    Dependency parentDependency = mockDependency("parent-lib", "3.0.0");
    // If the version does not match, the dependency must not be moved to parent layer
    Dependency nonMatchingParentDependency = mockDependency("parent-lib-different-version", "1.0.0");
```

Reply:
Ok, but a little bit different. foo and blah are both parent dependencies. So they are parent-lib1 and parent-lib2 now.